frontends/web/src/components/alert/Alert.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h } from 'preact';
+import i18n from '../../i18n/i18n';
+import Dialog from '../dialog/dialog';
+import { Button } from '../forms';
+
+var alerted: (message: string) => void;
+
+interface State {
+    message: string;
+    active: boolean;
+}
+
+interface Alert {
+    button: any;
+}
+
+class Alert extends Component<{}, State> {
+    constructor(props) {
+        super(props);
+        alerted = this.alerted;
+        this.state = {
+            message: '',
+            active: false,
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.handleKeyDown);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeyDown);
+    }
+
+    handleClose = e => {
+        this.setState({
+            active: false,
+            message: '',
+        });
+    }
+
+    handleKeyDown = e => {
+        if (e.keyCode === 13 && this.state.active) this.setState({ active: false });
+    }
+
+    alerted = message => {
+        this.setState({
+            message,
+            active: true,
+        }, () => {
+            this.button.base.focus();
+        });
+    }
+
+    public render({}, {
+        message,
+        active,
+    }) {
+        return active ? (
+            <Dialog>
+                {
+                    message.split('\n').map((line, i) => (
+                        <p
+                            key={i}
+                            class={ i === 0 ? 'first' : '' }>
+                            {line}
+                        </p>
+                    ))
+                }
+                <div class="buttons flex flex-row flex-end">
+                    <Button
+                        primary
+                        ref={ref => this.button = ref}
+                        onClick={this.handleClose}>
+                        {i18n.t('button.ok')}
```

Comment:
I always feel a bit uneasy with using `i18n.t` directly like this since the `@translate` decorator not only injects `t` as a property but also makes sure that the component is rerendered when the language is changed. I had this discussion with Eni in the past: It might be that it's enough if the uppermost element uses the translate decorator/higher-order component for rerendering. However, without verifying such a behavior (which might break with a future update of Preact), I wouldn't rely on it (as both React and Preact often preserve children with their state for performance reasons).

Reply:
@KasparEtter -- yes, this slipped through, will be updated.